segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/TrackedConnection.java

```diff
@@ -73,6 +78,11 @@ public void close() {
         this.connection.close();
     }
 
+    public void close(RunnableWithException callback) {
+        this.close();
+        Callbacks.invokeSafely(callback, ex -> log.error("Unable to run callback on TrackedConnection close.", ex));
```

Comment:
Tip: when logging errors, please include AS MUCH information as you can. The exception is mandatory, but is there anything else from this class that you may include? For example, we have a `toString` method that may return some useful info.